src/iso19111/crs.cpp

```diff
@@ -378,10 +378,16 @@ CRSNNPtr CRS::alterCSLinearUnit(const common::UnitOfMeasure &unit) const {
     {
         auto derivedProjCRS = dynamic_cast<const DerivedProjectedCRS *>(this);
         if (derivedProjCRS) {
+            auto cs = derivedProjCRS->coordinateSystem();
+            // DerivedProjectedCRS's factory accepts a generic CoordinateSystem,
+            // but does it make sense to consider any type other than Cartesian?
+            auto cartCS = util::nn_dynamic_pointer_cast<cs::CartesianCS>(cs);
```

Comment:
OK, but then there's no other possibility that requires the unit to be altered, right?

Reply:
According to https://docs.opengeospatial.org/as/18-005r4/18-005r4.html "Table 43: Defining elements of CoordinateSystems::DerivedProjectedCS class", the valid CS for a DerivedProjectedCRS are CartesianCS, CylindricalCS, OrdinalCS, PolarCS, SphericalCS.
CylindricalCS is a "three-dimensional coordinate system in Euclidean space in which position is specified by two linear coordinates and one angular coordinate", so in theory it could be affected, but there's no corresponding implementation of it in PROJ currently. The other CS use angular or non-linear units.

Reply:
Thanks for the reference. I was just looking at the classes in coordinatesystem.hpp and I didn't find any other with an `alterUnit` member function that was applicable, but I wasn't sure.